p2p/pex_reactor.go

```diff
@@ -108,18 +118,55 @@ func (r *PEXReactor) AddPeer(p Peer) {
 	} else { // For inbound connections, the peer is its own source
 		addr, err := NewNetAddressString(p.NodeInfo().ListenAddr)
 		if err != nil {
-			// peer gave us a bad ListenAddr. TODO: punish
+			// peer gave us a bad ListenAddr
+			r.MarkPeer(p, false, 1, Bad)
 			r.Logger.Error("Error in AddPeer: invalid peer address", "addr", p.NodeInfo().ListenAddr, "err", err)
 			return
 		}
+		r.MarkPeer(p, true, 1, Correct)
```

Comment:
AddPeer only runs once per connection, so not sure how useful this one is

Reply:
I'll remove it.